logging/config.go

```diff
@@ -140,25 +141,25 @@ func NewConfigFromMap(data map[string]string, components ...string) (*Config, er
 	}
 
 	lc.LoggingLevel = make(map[string]zapcore.Level)
-	for _, component := range components {
-		if ll := data["loglevel."+component]; len(ll) > 0 {
-			level, err := levelFromString(ll)
-			if err != nil {
-				return nil, err
+	for k, v := range data {
+		if strings.HasPrefix(k, "loglevel.") {
+			component := k[len("loglevel."):]
```

Comment:
1. const "loglevel."?
2. Move `component :=` into, it seems unused outside of it.

Reply:
I can't simply move things around as I don't want to index past the end of the string. But I can do:

```
if component := strings.TrimPrefix(k, "loglevel."); component != k {}
```

Would you like that?